contracts/PlasmaMVP.sol

```diff
@@ -87,34 +113,21 @@ contract RootChain is Ownable {
     // @param txnsPerBlock number of transactions per block
     // @param feesPerBlock amount of fees the validator has collected per block
     // @param blockNum     the block number of the first header
-    function submitBlock(bytes blocks, uint256[] txnsPerBlock, uint256[] feesPerBlock, uint256 blockNum)
+    function submitBlock(bytes32[] headers, uint256[] txnsPerBlock, uint256[] feesPerBlock, uint256 blockNum)
         public
-        onlyOwner
+        onlyOperator
     {
         require(blockNum == lastCommittedBlock + 1, "inconsistent block number ordering");
-        require(blocks.length > 0 && blocks.length % 32 == 0, "block roots must be of size 32 bytes");
```

Comment:
do we need the length > 0 check? I could see some strange things occurring without it, thought I'm not 100% sure

Reply:
actually nvm, assuming an empty array.length == 0 then it should be fine